src/network/mcpe/handler/InGamePacketHandler.php

```diff
@@ -657,6 +661,37 @@ public function handleBlockActorData(BlockActorDataPacket $packet) : bool{
 			}
 
 			$this->session->getLogger()->debug("Invalid sign update data: " . base64_encode($packet->nbt->getEncodedNbt()));
+		}elseif($block instanceof Beacon){
+			$inventory = $this->player->getCurrentWindow();
+			if(!($inventory instanceof BeaconInventory)){
+				return false;
+			}
+			$itemId = $inventory->getInput()->getId();
+			if(!isset(Beacon::ALLOWED_ITEM_IDS[$itemId])){
+				throw new PacketHandlingException("Invalid input $itemId");
+			}
+
+			if($block->getBeaconLevel() > 0 && $block->viewSky()){
+				$allowedEffects = $block->getAllowedEffect($block->getBeaconLevel());
+				$primaryEffect = EffectIdMap::getInstance()->fromId($nbt->getInt("primary", 0));
+				if($primaryEffect !== null && in_array($primaryEffect, $allowedEffects, true)){
+					$event = new BeaconActivateEvent($block, $primaryEffect);
+					$secondaryEffect = EffectIdMap::getInstance()->fromId($nbt->getInt("secondary", 0));
+					if($secondaryEffect !== null && in_array($secondaryEffect, $allowedEffects, true)){
+						$event->setSecondaryEffect($secondaryEffect);
+					}
+					$event->call();
+					if(!$event->isCancelled()){
+						$event->setPrimaryEffect($event->getPrimaryEffect());
+						$event->setSecondaryEffect($event->getSecondaryEffect());
+						$inventory->setInput($inventory->getInput()->pop());
```

Comment:
This is just going to set the item to itself. pop() returns the item removed from the stack.

Reply:
how am i supposed to remove the item from the beacon inventory? the item pop produces an error when we activate the beacon because it does not know the window with the id -10.

Reply:
```suggestion
						$item = $inventory->getInput();
						$item->pop();
						$inventory->setInput($item);
```

Reply:
> how am i supposed to remove the item from the beacon inventory? the item pop produces an error when we activate the beacon because it does not know the window with the id -10.

Too vague for me to have any idea what the problem is.

Reply:
Finally, the problem was solved, I was not using the pop() correctly